test/internal/laws/forall.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+var jsc = require('jsverify');
+var R = require('ramda');
+
+//  forall :: StrMap Function -> StrMap Function
+module.exports = function forall(laws) {
+  return Object.keys(laws).reduce(function(result, name) {
+    result[name] = function() {
+      var args = Array.prototype.slice.call(arguments);
+      function pred() {
+        var pair = laws[name].apply(null, arguments);
+        var lhs = pair[0];
+        var rhs = pair[1];
+        return R.equals(lhs, rhs);
```

Comment:
almost yes, but if this should be moved to fl-laws then I think the R.equals should not be hardcoded here, also some functions don't need the `eq` (Setoid), I think the "laws.js" should have api something like [this](https://github.com/fantasyland/fantasy-land/pull/195#issuecomment-256894273):
```js
{
  Setoid: () => ({ ... }),
  Semigroup: (eq) => ({ ... }),
  Applicative: (T, eq) => ({ ... }),
}
```